pkg/promclient/promclient.go

```diff
@@ -336,18 +376,13 @@ func (c *Client) QueryInstant(ctx context.Context, base *url.URL, query string,
 
 	level.Debug(c.logger).Log("msg", "querying instant", "url", u.String())
 
-	req, err := http.NewRequest("GET", u.String(), nil)
-	if err != nil {
-		return nil, nil, errors.Wrap(err, "create GET request")
-	}
+	span, ctx := tracing.StartSpan(ctx, "/prom_query_instant HTTP[client]")
+	defer span.Finish()
 
-	req = req.WithContext(ctx)
-
-	resp, err := c.httpClient.Do(req)
+	body, _, err := c.get2xx(ctx, &u)
 	if err != nil {
-		return nil, nil, errors.Wrapf(err, "perform GET request against %s", u.String())
+		return nil, nil, err
```

Comment:
no wrapping is necessary any more?

Reply:
correct

Reply:
https://github.com/thanos-io/thanos/pull/2277